src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -672,6 +673,15 @@ private void GeneratePropMetadataInitFunc(SourceWriter writer, string propInitMe
                         writer.WriteLine($"properties[{i}].Order = {property.Order};");
                     }
 
+                    if (property.IsGetterNonNullableAnnotation)
+                    {
+                        writer.WriteLine($"properties[{i}].IsGetNullable = false;");
+                    }
+                    if (property.IsSetterNonNullableAnnotation)
+                    {
+                        writer.WriteLine($"properties[{i}].IsSetNullable = false;");
+                    }
```

Comment:
This isn't accounting for the opt-out and it should be defaulted to true.
```suggestion

                    if (property.IsGetterNonNullableAnnotation)
                    {
                        writer.WriteLine($"properties[{i}].IsGetNullable = !{OptionsLocalVariableName}.RespectNullableAnnotations;");
                    }
                    else
                    {
                        writer.WriteLine($"properties[{i}].IsGetNullable = true;");
                    }
```

Reply:
I've changed the semantics of the properties slightly per https://github.com/dotnet/runtime/issues/100144#issuecomment-2122868748: the values of `IsGetNullable` and `IsSetNullable` always reflect the nullability annotations regardless of the state of the global flag (so it can always be read as metadata, which is useful in the case of JSON schema extraction). But these only have effect in serialization if the global setting has been opted into.

Secondly, the value of `JsonPropertyInfo<T>.IsGetNullable` and `IsSetNullable` default to `default(T) is null`. It therefore only needs to be set by the source generator in cases where you have a non-nullable reference type, where it explicitly needs to be set to `false`.